modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -501,26 +501,51 @@ bool GDMono::_load_project_assembly() {
 #endif
 
 #ifdef GD_MONO_HOT_RELOAD
+void GDMono::reload_failure() {
+	if (++project_load_failure_count >= 5) {
+		// After reloading a project has failed 5 times in a row, update the path and modification time
+		// to stop any further attempts at loading this assembly, which probably is never going to work anyways
+		project_load_failure_count = 0;
+
+		ERR_PRINT_ED(".NET: Giving up on assembly reloading. Please restart the editor if unloading was failing.");
+
+		String assembly_name = GLOBAL_GET("dotnet/project/assembly_name");
+		if (assembly_name.is_empty()) {
+			assembly_name = ProjectSettings::get_singleton()->get_safe_project_name();
+		}
+		String assembly_path = GodotSharpDirs::get_res_temp_assemblies_dir().path_join(assembly_name + ".dll");
+		assembly_path = ProjectSettings::get_singleton()->globalize_path(assembly_path);
+		project_assembly_path = assembly_path.simplify_path();
+		project_assembly_modified_time = FileAccess::get_modified_time(assembly_path);
+	}
+}
+
 Error GDMono::reload_project_assemblies() {
 	ERR_FAIL_COND_V(!runtime_initialized, ERR_BUG);
 
 	finalizing_scripts_domain = true;
 
-	CSharpLanguage::get_singleton()->_on_scripts_domain_about_to_unload();
-
 	if (!get_plugin_callbacks().UnloadProjectPluginCallback()) {
-		ERR_FAIL_V_MSG(Error::FAILED, ".NET: Failed to unload assemblies.");
+		ERR_PRINT_ED(".NET: Failed to unload assemblies.");
```

Comment:
If we're going to retry loading multiple times, I feel like showing errors while we're still trying may be confusing to users. If reloading is able to succeed after multiple tries, then maybe we should wait until all the tries fail to show an error.

Currently with all these `ERR_PRINT_ED`, in the worst case scenario you get 6 error messages and that may be excessive.

Reply:
The 5 errors caused by this will get folded into a single popup so you really only get 2 editor popups.

Reply:
You are right, and even though they all show individually in the output dock, it's probably fine.